deployments/helm-chart/templates/controller-service.yaml

```diff
@@ -29,14 +29,24 @@ spec:
 {{- end }}
   type: {{ .Values.controller.service.type }}
   ports:
-  - port: 80
+{{- if .Values.controller.service.httpPort.enable }}
+  - port: {{ .Values.controller.service.httpPort.port }}
     targetPort: 80
     protocol: TCP
     name: http
-  - port: 443
+  {{- if (eq .Values.controller.service.type "NodePort") }}
+    nodePort: {{ .Values.controller.service.httpPort.nodePort }}
+    {{- end }}
+  {{- end }}
+{{- if .Values.controller.service.httpsPort.enable }}
+  - port: {{ .Values.controller.service.httpsPort.port }}
     targetPort: 443
     protocol: TCP
     name: https
+  {{- if (eq .Values.controller.service.type "NodePort") }}
+    nodePort: {{ .Values.controller.service.httpsPort.nodePort }}
+    {{- end }}
+  {{- end }}
```

Comment:
Same as lines 39, 40, 48, 49. Please make sure the indentation of each `{{- end }}` is the same as of the corresponding `{{- if }}`